cranelift/codegen/src/isa/x64/lower/isle.rs

```diff
@@ -940,6 +954,79 @@ impl Context for IsleContext<'_, '_, MInst, X64Backend> {
     fn box_synthetic_amode(&mut self, amode: &SyntheticAmode) -> BoxSyntheticAmode {
         Box::new(amode.clone())
     }
+
+    ////////////////////////////////////////////////////////////////////////////
+    ///// External assembler methods.
+    ////////////////////////////////////////////////////////////////////////////
+
+    fn is_imm8(&mut self, src: &GprMemImm) -> Option<AssemblerImm8> {
+        match src.clone().to_reg_mem_imm() {
+            RegMemImm::Imm { simm32 } if simm32 <= u8::MAX as u32 => {
+                Some(AssemblerImm8::new(simm32 as u8))
+            }
+            _ => None,
+        }
+    }
```

Comment:
Question on this: how does this know that an unsigned value is desired? For example if you have `(a: u8) & -2` that means it would fail `is_imm8`, right? But the x64 instruction should be able to encode that?

With the assembler immediates all annotated with if they're sign/zero-extended should there be `Simm8` and `Uimm8` and the extractors here would be `is_simm8` and `is_uimm8`?

(also as a side note I find `u8::try_from(...)` handy in cases of conversion like this so you don't have to do any `as` casts yourself)

Reply:
Ok, I see what you mean. The assembler doesn't care too much about the sign, it just wants 8 bits, so I chose to use `u8` to pass those along. What I was trying to do here (incorrectly) is check that we can fit whatever was in the `simm32: i32` into those 8 bits.